packages/frontend-web/src/app/scenes/Settings/components/RuleRow/RuleRow.tsx

```diff
@@ -101,8 +101,8 @@ export interface IRuleRowProps {
   selectedCategory: string;
   selectedTag?: string;
   onDelete?(rule: IRuleModel | ITaggingSensitivityModel | IPreselectModel): any;
-  onCategoryChange?(value: number): any;
-  onTagChange?(value: number): any;
+  onCategoryChange?(value: string): any;
```

Comment:
(Caveat: I have no idea how TypeScript works.)

Does this cause a problem if the categoryId is a number?

Reply:
I don't believe so. And the `categoryId` should always be a string anyway. 

When I first started looking into this issue, I made all the `number` -> `string` type changes and it didn't fix the problem of the `categoryId` getting cast as an integer. It also didn't throw any compilation errors, which I can't say I understand. The line of code in this PR that fixes the root problem is changing:

```javascript
callback(parseInt(e.target.value, 10));
```
to 
```javascript
callback(e.target.value, 10);
```